packages/tpetra/core/test/CrsMatrix/CrsMatrix_UnitTests2.cpp

```diff
@@ -430,6 +431,20 @@ inline void tupleToArray(Array<T> &arr, const tuple &tup)
     } else {
       TEST_COMPARE_FLOATING_ARRAYS(norms,zeros,MT::zero());
     }
+
+    // test the constructors based on 4 maps + local matri_crsx
+    RCP<MAT> tri_crs_2 = rcp(new MAT(tri_crs->getRowMap(), tri_crs->getColMap(), tri_crs->getDomainMap(),
+            tri_crs->getRangeMap(), tri_crs->getLocalMatrix()) );
+    // test the action
```

Comment:
Should also test whether the resulting matrix is fill complete, and whether the Maps and Import (and Export, if applicable) objects match.